pkg/serverutils/asynccache/asynccache.go

```diff
@@ -33,20 +32,31 @@ func NewAsyncCache[T any](ctx context.Context, reloadPeriod time.Duration, cachi
 		cachingFunc:  cachingFunc,
 	}
 
-	var err error
-	wait.PollUntilContextTimeout(
+	err := wait.PollUntilContextTimeout(
 		ctx,
 		initializationRetryInterval,
 		initializationTimeout,
-		initializeImmediately,
-		func(ctx context.Context) (bool, error) {
-			item, err := cachingFunc(ctx)
-			if err != nil {
-				klog.V(4).Infof("failed to setup an async cache (retrying in %v) - caching func returned error: %v", initializationRetryInterval, err)
-				return false, nil
+		true,
+		func(pollingCtx context.Context) (bool, error) {
+			itemChan := make(chan struct{}, 1)
+			var item T
+			var err error
+			go func() {
+				item, err = cachingFunc(ctx)
+				itemChan <- struct{}{}
+			}()
```

Comment:
Wondering, could we use channels to pass both item and error through to be handled below:
```suggestion
		func(pollingCtx context.Context) (bool, error) {
			itemChan := make(chan T, 1)
			errChan := make(chan error, 1)
			go func() {
				i, e := cachingFunc(ctx)
				if e != nil {
				  errChan <- e
				  return
				}
				itemChan <- i
			}()
```

Reply:
We could although I am not sure it's necessary at this moment. The channel simply serves as a notifier that a result is available.